workspaces/ui-v2/src/components/HttpBodySelector.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import {
+  FormControl,
+  FormControlLabel,
+  makeStyles,
+  Radio,
+  RadioGroup,
+} from '@material-ui/core';
+
+import { FontFamily } from '<src>/styles';
+
+type HttpBodySelectorProps<T> = {
+  items: T[];
+  children: (body: T) => React.ReactNode;
+  getDisplayName: (item: T) => string;
+};
+
+/**
+ * A component that is used to select / handle different items
+ */
+export const HttpBodySelector: <T extends {}>(
+  props: HttpBodySelectorProps<T>
+) => React.ReactElement = ({ items, children, getDisplayName }) => {
+  const [selectedItem, setSelectedItem] = useState(0);
+  const inBoundsSelected = Math.min(selectedItem, items.length - 1);
+  const classes = useStyles();
+
+  return (
+    <>
+      {items.length === 0 ? (
+        <>No bodies found</>
```

Comment:
I don't think we ever hit this branch, but since it's a "generic" component (likely we could rename this to MultiSelector) - we would hit it and handle it. It's poorly styled because we don't hit this path